sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,410 @@
+
+
+class KafkaValidatorThread(KafkaThreadBase):
+    """
+    Thread to validate that records added to Kafka are correctly written to ScyllaDB.
+    """
+
+    def __init__(self, tester, params: SCTConfiguration, timeout: int, **kwargs):
+        super().__init__(tester, params, timeout, **kwargs)
+        self.mismatches = []
+        self.batch_size = int(kwargs.get('batch-size', 1000))
+        self.keyspace = kwargs.get('keyspace', None)
+        self.table = kwargs.get('table', None)
+        self.current_key = 1
+        self.batch_wait_time = 10
+        self.batch_poll_interval = 2
+        self.validated_rows = 0
+
+    def run(self) -> None:
+        while not self.termination_event.is_set():
+            batch_end = min(self.current_key + self.batch_size - 1, self.num_records)
+            batch_size = batch_end - self.current_key + 1
+            batch_start_time = time.time()
+            while True:
+                rows = self.read_rows_from_scylla(self.current_key, batch_end)
+                if len(rows) < batch_size:
+                    if time.time() - batch_start_time > self.batch_wait_time:
+                        error_msg = (f"Timeout waiting for records from {self.current_key} to {batch_end} "
+                                     f"to be in ScyllaDB. Expected {batch_size}, found {len(rows)}")
+                        LOGGER.error(error_msg)
+                        self.mismatches.append(error_msg)
+                        break  # move to next batch
+                    LOGGER.debug("Only %s out of %s records found in ScyllaDB for keys %s to %s. "
+                                 "Waiting...", len(rows), batch_size, self.current_key, batch_end)
+                    time.sleep(self.batch_poll_interval)
+                    continue
+                self.compare_rows(rows)
+                break  # move to next batch
+            self.current_key = batch_end + 1
+
+            if self.current_key >= self.num_records:
+                LOGGER.info("All records processed, stopping Kafka validator thread.")
+                self.stop()
+
+    def read_rows_from_scylla(self, range_start: int, range_end: int) -> list:
+        LOGGER.info("Reading rows from ScyllaDB for keys %s to %s.", range_start, range_end)
+        with self.tester.db_cluster.cql_connection_patient(self.tester.db_cluster.nodes[0]) as session:
+            query = f"SELECT * FROM {self.keyspace}.{self.table} WHERE id >= %s AND id <= %s ALLOW FILTERING"
+            rows = list(session.execute(query, parameters=(range_start, range_end)))
+            LOGGER.debug("Retrieved %d rows from ScyllaDB for the current batch.", len(rows))
+            return rows
+
+    def compare_rows(self, rows: list) -> None:
+        LOGGER.info("Comparing batch of ScyllaDB rows with expected data.")
+        for row in rows:
+            id_value = row.id
+            expected_data = self.generate_data_from_schema(json.loads(self.value_schema), id_value)
```

Comment:
please use explicit arugments names to make it clear that `id_value` is seed

Reply:
Fixed this